Orion/Orion/Models/Product.m

```diff
@@ -7,37 +7,89 @@
 
 #import "Product.h"
 
+static bool productAPICheck = false;
 //TODO: Add dicitionary product method
 @implementation Product
 
 - (instancetype)initWithDictionary:(NSDictionary *)dictionary {
     self = [super init];
     //sets up intializer for initWithDictionary
     
-    if(self.name == NULL)
-    {
-        if (self) {
-            self.name = dictionary[@"names"][@"title"];
-            self.productImage = dictionary[@"images"][@"standard"];
-            self.productSKU = dictionary[@"sku"];
-            self.productPrice = dictionary[@"prices"][@"regular"];
+    
+    if (self) {
+        self.name = dictionary[@"names"][@"title"];
+        self.productImage = dictionary[@"images"][@"standard"];
+        self.productSKU = dictionary[@"sku"];
+        self.productPrice = dictionary[@"prices"][@"regular"];
     }
     
+//    if(productAPICheck == false)//initializes based on which API is being used
+//    {
+//        if (self) {
+//            self.name = dictionary[@"names"][@"title"];
+//            self.productImage = dictionary[@"images"][@"standard"];
+//            self.productSKU = dictionary[@"sku"];
+//            self.productPrice = dictionary[@"prices"][@"regular"];
+//    }
+//    else
+//    {
+//        if (self) {
+//            self.name = dictionary[@"name"];
+//            self.productImage = dictionary[@"image"];
+//            self.productSKU = dictionary[@"sku"];
+//            self.productPrice = dictionary[@"regularPrice"];
+//    }
+//    }
+    
         //...
     // Initialize any other properties
+    return self;
+}
+
+- (instancetype)initWithProductAPIDictionary:(NSDictionary *)dictionary {
+    self = [self init];
+    
+    if(self)  {
+        self.name = dictionary[@"name"];
+        self.productImage = dictionary[@"image"];
+        self.productSKU = dictionary[@"sku"];
+        self.productPrice = dictionary[@"regularPrice"];
     }
     return self;
 }
 
+//- (instancetype)addedWithProductSpec:(Product *)dictionary {
+//    self = [self init];
+//    
+//    if(self) {
+//        
+//    }
+//}
+
 
 + (NSMutableArray *)productsWithArray:(NSArray *)dictionaries {
     NSMutableArray *products = [NSMutableArray array];
     for (NSDictionary *dictionary in dictionaries) {
+        productAPICheck = false;
         Product *product = [[Product alloc] initWithDictionary:dictionary];
         [products addObject:product];
     }
     return products;
 }
+
++ (NSMutableArray *)productsWithProductAPIArray: (NSArray *)dictionaries {
+    NSMutableArray *products = [NSMutableArray array];
+    for (NSDictionary *dictionary in dictionaries) {
+        productAPICheck = true;
+        Product *product = [[Product alloc] initWithProductAPIDictionary:dictionary];
+        [products addObject:product];
+    }
+    return products;
+}
+
+//+ (Product *)detailProductAdditions: (Product *)sentProduct {
```

Comment:
ditto, to avoid spam i'll stop mentioning these here. But good to do a quick scan over "//" and"/*" instances and clear out non-descriptive comments